yaml/build.go

```diff
@@ -63,22 +65,31 @@ func handleBuildArgs(value interface{}) (map[string]string, error) {
 	case map[interface{}]interface{}:
 		return handleBuildArgMap(v)
 	case []interface{}:
-		args = map[string]string{}
-		for _, elt := range v {
-			uniqArgs, err := handleBuildArgs(elt)
-			if err != nil {
-				return args, nil
-			}
-			for mapKey, mapValue := range uniqArgs {
-				args[mapKey] = mapValue
-			}
-		}
-		return args, nil
+		return handleBuildArgSlice(v)
 	default:
 		return args, fmt.Errorf("Failed to unmarshal Build args: %#v", value)
 	}
 }
 
+func handleBuildArgSlice(s []interface{}) (map[string]string, error) {
+	var args = map[string]string{}
+	for _, arg := range s {
+		// check if a value is provided
+		switch v := strings.SplitN(arg.(string), "=", 2); len(v) {
+		case 1:
+			// we don't use EnvironmentLookup here as docker does not use values in the env_file or environment config stanza for this
+			envValue := os.Getenv(v[0])
```

Comment:
The problem with using `os.Getenv` is that it can't be overridden to perform lookups in a customized way. We have an [existing structure](https://github.com/docker/libcompose/blob/master/config/types.go#L10) that solves this problem.

`EnvironmentLookup` is only available in the `Project` struct, so I think some of the processing code will have to be moved to the `project` package. Specifically, we do processing for the `environment` key in `CreateService`, so it probably makes sense for this to be in `CreateService` as well.